### ISSUE TITLE
Run Action in a Docker Container that is not the same container as the runner

### ISSUE THREAD

**Issue by shicholas** [cypress-io/github-action#222]
It would be awesome to be able to configure this action like this:

```
      - name: Run Cypress tests for the Interface
        uses: cypress-io/github-action@v2
        with:
          working-directory: ./packages/interface
          container: cypress/browsers:node12.13.0-chrome78-ff70
```

Specifying the container in the job itself is different than the current [Docker Image](https://github.com/cypress-io/github-action#docker-image) documentation because there would be two different environments for the Cypress step and the rest of the steps in the job workflow. The biggest benefit for us is that we wouldn't need a `yarn install` step because everything is already pre-installed in the Docker container, and we could technically start a web server with non-node processes, like using Hugo.

**Comment by bahmutov** [comment]
Can you provide full example of what you are trying to do? Maybe a pull request with one more example subfolder showing the desired output.

**Comment by shicholas** [comment]
Currently, we run Cypress in CI with [this code](https://github.com/NeonLaw/codebase/blob/development/.github/workflows/continuous_integration.yml#L65-L69) instead of GitHub action. Since we use `docker-compose` to run the rest of our tests, we can't use the GitHub Action provided here with your supported Docker containers because that would necessitate that the host machine would be a Cypress docker container instead of a regular Ubuntu box, and the former does not support Docker-in-Docker.

Does this make sense?

**Comment by bahmutov** [comment]
How would the action's source code change to support this? I will add "help wanted" to this issue, since it seems like a very narrow use case, and I would love to have an external PR implementing this.

**Comment by shicholas** [comment]
I'm not sure, I changed my code to use the regular GitHub Action as this better supports adding additional dependencies like `cypress-axe`, so I opted not to go down this route anymore. Instead, [I run `docker-compose up` and then `cypress` on the same host machine where `docker` is running](https://github.com/NeonLaw/codebase/blob/development/.github/workflows/continuous_integration.yml#L45-L59). Thanks for looking into this!
